xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp

```diff
@@ -1083,9 +1085,7 @@ void CRenderManager::PrepareNextRender()
   double frameOnScreen = m_dvdClock.GetClock();
   double frametime = 1.0 / g_graphicsContext.GetFPS() * DVD_TIME_BASE;
 
-  // correct display latency
-  // internal buffers of driver, assume that driver lets us go one frame in advance
-  double totalLatency = DVD_SEC_TO_TIME(m_displayLatency) - DVD_MSEC_TO_TIME(m_videoDelay) + 2* frametime;
+  double totalLatency = DVD_SEC_TO_TIME(TotalLatency());
```

Comment:
I'm aware of that distinction in principle, but it's very easy to get confused here...

fwiw on Wayland with mesa and usual compositors you will probably lose at least one frame in your example of blocking for ~40 ms. Even if you render very fast afterwards and immediately submit the two rendered frames to the compositor, it will typically just discard the earlier buffer if it hasn't used it yet. Also, mesa synchronizes with the compositor redraw cycle by blocking in eglSwapBuffers when eglSwapInterval(1) is used, so you cannot render faster than vrefresh in that case.

Sorry for the sidetracking - let's get back to the main issue of correcting the latency so this moves forward :-) Should I take the time from `PresentRender` at function exit or enter, i.e. before or after `eglSwapBuffers` (makes a big difference)? And is this similar to https://github.com/FernetMenta/kodi-agile/blob/master/xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp#L1097 or something else altogether?